mempool/src/mempool.rs

```diff
@@ -89,17 +99,24 @@ impl Mempool {
                 config.filter_limit,
             ))),
             executor_handle: Mutex::new(None),
+            control_executor_handle: Mutex::new(None),
         }
     }
 
-    /// Starts the mempool executor
+    /// Starts the mempool executors
     ///
     /// Once this function is called, the mempool executor is spawned.
     /// The executor will subscribe to the transaction topic from the the network.
-    pub async fn start_executor<N: Network>(&self, network: Arc<N>, monitor: Option<TaskMonitor>) {
+    pub async fn start_executor<N: Network>(
+        &self,
+        network: Arc<N>,
+        monitor: Option<TaskMonitor>,
+        ctrl_monitor: Option<TaskMonitor>,
+    ) {
         let mut executor_handle = self.executor_handle.lock().await;
+        let mut control_executor_handle = self.control_executor_handle.lock().await;
 
-        if executor_handle.is_some() {
+        if executor_handle.is_some() && control_executor_handle.is_some() {
             // If we already have an executor running, don't do anything
             return;
         }
```

Comment:
Both executors are created/terminated in tandem. If for some reason one of them crashes we don't have a way to recreate it (this applies to the current mempool as well, if the executor dies we are not monitoring its health and this would indicate a major problem in our code)
If a node has a mempool, it will have both a control and regular executors.